web_poet/serialization/api.py

```diff
@@ -135,7 +136,19 @@ def serialize(deps: Iterable[Any]) -> SerializedData:
         cls = dep.__class__
         if is_injectable(cls):
             raise ValueError(f"Injectable type {cls} passed to serialize()")
-        result[_get_name_for_class(cls)] = serialize_leaf(dep)
+        if cls is AnnotatedResult:
+            key = f"AnnotatedResult {_get_name_for_class(dep.result.__class__)}"
+        else:
+            key = _get_name_for_class(cls)
+
+        if key in result:
```

Comment:
It seems this condition won't be True if there is Annotated and non-Annotated version of the same dependency. Is it an issue?

Reply:
Annotated and non-Annotated versions of the same dependency can be serialized together (the key is different for the Annotated classes) so that should be fine.